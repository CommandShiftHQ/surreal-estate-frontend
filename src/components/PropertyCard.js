import React from "react";
import PropTypes from "prop-types";

const PropertyCard = ({
  _id,
  title,
  type,
  city,
  bathrooms,
  bedrooms,
  price,
  email,
  userID,
  onSaveProperty,
}) => {
  return (
    <div className="PropertyCard">
      <header>
        <h2>{title}</h2>
        <h3>{`${type} - ${city}`}</h3>
        {userID && (
          <button type="button" onClick={() => onSaveProperty(_id)}>
            Save
          </button>
        )}
      </header>
      <section>
        <div>{`Bathrooms: ${bathrooms}`}</div>
        <div>{`Bedrooms: ${bedrooms}`}</div>
        <div>{`Price: Â£ ${price}`}</div>
      </section>
      <footer>
        <a href={`mailto:${email}`}>Email</a>
      </footer>
    </div>
  );
};

PropertyCard.propTypes = {
  _id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  bathrooms: PropTypes.string.isRequired,
  bedrooms: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  userID: PropTypes.string.isRequired,
  onSaveProperty: PropTypes.func.isRequired,
};

export default PropertyCard;
