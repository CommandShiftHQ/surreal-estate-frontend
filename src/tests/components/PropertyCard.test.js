import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

const props = {
  title: "2 bedroom period property",
  type: "Flat",
  bathrooms: "2",
  bedrooms: "2",
  price: "1000",
  city: "Manchester",
  email: "myemail@email.com",
};

// test("renders correctly", () => {
//   const { asFragment } = render(<PropertyCard />);

//   expect(asFragment()).toMatchSnapshot();
// });

test("displays a title", async () => {
  const { findByText } = render(<PropertyCard {...props} />);
  const title = await findByText(props.title);

  expect(title.textContent).toBe(props.title);
});

test("displays a type with a location", async () => {
  const { findByText } = render(<PropertyCard {...props} />);
  const typeAndCityText = `${props.type} - ${props.city}`;
  const typeAndLocationNode = await findByText(typeAndCityText);

  expect(typeAndLocationNode.textContent).toBe(typeAndCityText);
});

test("displays a number of bathrooms", async () => {
  const { findByText } = render(<PropertyCard {...props} />);
  const bathroomsText = `Bathrooms: ${props.bathrooms}`;
  const bathroomsNode = await findByText(bathroomsText);

  expect(bathroomsNode.textContent).toBe(bathroomsText);
});

test("displays a number of bedrooms", async () => {
  const { findByText } = render(<PropertyCard {...props} />);
  const bedroomsText = `Bedrooms: ${props.bedrooms}`;
  const bedroomsNode = await findByText(bedroomsText);

  expect(bedroomsNode.textContent).toBe(bedroomsText);
});

test("displays a price", async () => {
  const { findByText } = render(<PropertyCard {...props} />);
  const priceText = `Price: Â£ ${props.price}`;
  const priceNode = await findByText(priceText);

  expect(priceNode.textContent).toBe(priceText);
});

test("can send an email", async () => {
  const { findByText } = render(<PropertyCard {...props} />);
  const email = await findByText("Email");

  expect(email.closest("a").href).toBe(`mailto:${props.email}`);
});
