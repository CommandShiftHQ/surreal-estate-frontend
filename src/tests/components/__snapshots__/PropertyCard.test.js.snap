// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="PropertyCard"
  >
    <header>
      <h2>
        2 bedroom period property
      </h2>
      <h3>
        Flat - Manchester
      </h3>
      <button
        type="button"
      >
        Save
      </button>
    </header>
    <section>
      <div>
        Bathrooms: 2
      </div>
      <div>
        Bedrooms: 2
      </div>
      <div>
        Price: Â£ 1000
      </div>
    </section>
    <footer>
      <a
        href="mailto:myemail@email.com"
      >
        Email
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
